yearn/treasury/treasury.py

```diff
@@ -86,15 +89,26 @@ class Treasury:
     def __init__(self, watch_events_forever=False):
         self.addresses = list(TREASURY_WALLETS)
         self._transfers = []
-        self._topics_in = [
-            '0xddf252ad1be2c89b69c2b068fc378daa952ba7f163c4a11628f55a4df523b3ef',
-            None,
-            ['0x000000000000000000000000' + address[2:] for address in self.addresses],
-        ]  # Transfers into Yearn wallets
-        self._topics_out = [
-            '0xddf252ad1be2c89b69c2b068fc378daa952ba7f163c4a11628f55a4df523b3ef',
-            ['0x000000000000000000000000' + address[2:] for address in self.addresses],
-        ]  # Transfers out of Yearn wallets
+        self._start_block = 10502337 # Treasury didn't exist prior to block 10502337
+
+        # define transfer signatures for Transfer events from ERC-20 and ERC-677 contracts
+        transfer_sigs = [
+            ERC20_TRANSFER_EVENT_HASH,
+            ERC677_TRANSFER_EVENT_HASH
+        ]
+        treasury_addresses = ['0x000000000000000000000000' + address[2:] for address in self.addresses]
+        self._topics = [
```

Comment:
this part is great, didn't know it was possible

Reply:
using encode_* helper methods like you proposed